beniget/beniget.py

```diff
@@ -1113,27 +1201,107 @@ def visit_excepthandler(self, node):
         return dnode
 
     def visit_arguments(self, node):
-        for arg in node.args:
+        for arg in _iter_arguments(node):
             self.visit(arg)
-
-        for arg in node.posonlyargs:
-            self.visit(arg)
-
-        if node.vararg:
-            self.visit(node.vararg)
-
-        for arg in node.kwonlyargs:
-            self.visit(arg)
-        if node.kwarg:
-            self.visit(node.kwarg)
-
+            
     def visit_withitem(self, node):
         dnode = self.chains.setdefault(node, Def(node))
         self.visit(node.context_expr).add_user(dnode)
         if node.optional_vars:
             self.visit(node.optional_vars)
         return dnode
 
+def _iter_arguments(args):
```

Comment:
I like this simplification. Combined with the doc update etc, you could extract them to a separate commit, it would make review easier for me.